File: spring-app/src/main/java/br/com/vitortccappspring/tccvitor/models/Cliente.java
package br.com.vitortccappspring.tccvitor.models;


import jakarta.persistence.*;

import java.io.Serializable;
import java.util.Date;
import java.util.UUID;

@Entity
@Table(name="tb_cliente")
public class Cliente implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID uuid;
    private String primeiroNome;
    private String sobreNome;
    private Date dataDeCadastro;

    private String telefone;

    private String email;

    private String dataNascimento ;

    public UUID getUuid() {
        return uuid;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDataNascimento() {
        return dataNascimento;
    }

    public void setDataNascimento(String dataNascimento) {
        this.dataNascimento = dataNascimento;
    }


    public String getPrimeiroNome() {
        return primeiroNome;
    }

    public void setPrimeiroNome(String primeiroNome) {
        this.primeiroNome = primeiroNome;
    }

    public String getSobreNome() {
        return sobreNome;
    }

    public void setSobreNome(String sobreNome) {
        this.sobreNome = sobreNome;
    }

    public Date getDataDeCadastro() {
        return dataDeCadastro;
    }

    public void setDataDeCadastro(Date dataDeCadastro) {
        this.dataDeCadastro = dataDeCadastro;
    }


}

File: node-app/src/entity/Cliente.ts

import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity({ name: 'tb_cliente' })
export class Cliente {

    @PrimaryGeneratedColumn("uuid")
    uuid: string;

    @Column({ name: "primeiro_nome" })
    primeiroNome: string;

    @Column({ name: "sobre_nome" })
    sobreNome: string;

    @Column({ type: 'timestamp', name: "data_de_cadastro" })
    dataDeCadastro: Date;


    @Column({ name: "telefone" })
    telefone: string;

    @Column({ name: "email" })
    email: string;

    @Column({ name: "data_nascimento" })
    dataNascimento: string;


}
